### PR TITLE
Visual improvements to backgrounds

### PR DESCRIPTION
**Pull request by DerpyChap**
This PR introduces some visual improvements to backgrounds.

Image backgrounds:
 - Images are now scaled to fit the screen
   - This not only ensures that low resolution images fill the frame, it also lets people use higher resolution images than ~1780×1000 without them being cropped
 - Images that aren't 16:9 will now change the camera's background colour to the average colour of the image

Empty backgrounds:
 - These will now use the Free Play background by default, which should look better than the plain grey background that was used before
 - Users can set what's used for the empty background in `TrombLoader.cfg` with the following options: `freeplay`, `freeplay-static`, `grey` and `black`

Video backgrounds:
 - The playback speed of video backgrounds is now adjusted when Practice Mode or Turbo Mode are in use to keep things in sync with the audio
 - Since there's an increased chance of flashing imagery when using Turbo Mode, a config option has been added that will force an image or empty background to load when Turbo Mode is enabled
 - TootTally has its own playback speed modifier and will need to be patched separately in order for the video speed to change

This PR changes the hijacked background from Ball Game to Free Play, which was mostly done to make the empty background changes a bit easier. The Free Play background has the same structure as every other background and still includes an audio controller, so nothing should break from this change.

Also, Turbo Mode is still broken with Smooth Scrolling (unrelated to this PR). This is apparently related to TrombLoader or BaboonAPI but I've no idea what's causing it. I've considered adding a workaround for this that forces the game to use Time-Locked scrolling when Turbo Mode is enabled but I'll wait for feedback first.

### CODE REVIEW

**Comment by hedgehog1029** [comment]
... uhh hang on

**Comment by hedgehog1029** [comment]
okay there we go. Moved to 2.4.0 to not block 2.3.0 which was wayyyyyyyy overdue a release. I wanna get this out as a beta to charters so we can test a wide variety of charts and make sure we haven't broken anything obvious.

File: CustomTracks/Backgrounds/ImageBackground.cs [hunk 1]
```diff
@@ -17,10 +17,67 @@ public override void SetUpBackground(BGController controller, GameObject bg)
...
+        renderer.transform.localScale = Vector3.one * scaleFactor;
+
+
+        // just gives a bit of wiggle room for images that aren't quite 16:9
+        if (aspectRatio >= 1.7f && aspectRatio <= 1.78f) {
+            camera.backgroundColor = Color.black;
+        }
+        else
+        {
+            System.Diagnostics.Stopwatch stopwatch = new System.Diagnostics.Stopwatch();
```
Comment (hedgehog1029) [inline CustomTracks/Backgrounds/ImageBackground.cs]: :rotating_light: leftover debug line detected

**Comment by DerpyChap** [comment]
sounds good to me 👍

**Review by hedgehog1029 — CHANGES_REQUESTED** [review]

File: CustomTracks/Backgrounds/ImageBackground.cs (hunk 1 above)
Comment (DerpyChap) [reply CustomTracks/Backgrounds/ImageBackground.cs]: 🙈

**Comment by hedgehog1029** [comment]
oh god has it been almost a year since you opened this. how

this is accepted for 2.4.0 just getting a beta out before release!

**Pull request merged**